File: secondary_implementations/python/modules/FacebookModule.py
# vim: ts=2 sw=2 noexpandtab
import requests
from .AbstractModule import AbstractModule


class FacebookModule(AbstractModule):
	def try_login(self, uname, password):
		headers = {'user-agent': AbstractModule().user_agent()}
		resp = requests.get('https://www.facebook.com', headers=headers)

		payload = {'email': uname, 'pass': password}
		loginResp = requests.post(
			'https://www.facebook.com/login.php', data=payload, headers=headers,
			cookies=resp.cookies, allow_redirects=False)
		
		return loginResp.status_code == 302


facebook = FacebookModule()

File: secondary_implementations/ruby/modules/FacebookModule.rb
require_relative 'AbstractModule'  

class FacebookModule < AbstractModule
  def try_login(username, password) 
  	headers = { 'user-agent' => user_agent}
  	response = RestClient.get("https://www.facebook.com", {:headers => headers}){|response, request, result| response }

  	payload = {'email' => username, 'pass' => password}
  	response = RestClient.post("https://www.facebook.com/login.php", payload, {:headers => headers, :cookies => response.cookies, :max_redirects => 0}){|response, request, result| response }

  	return response.code == 302
  end  
end  